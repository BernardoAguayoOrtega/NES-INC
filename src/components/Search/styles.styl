.search i
  transform: scaleX(-1)

.searchformOff
  display: none
  margin-right: 10px

.searchform
  display: inline-block
  margin-right: 10px

  input
    height:28px

.slide_search-left
  animation: slide_search-left 0.5s

@keyframes slide_search-left
  0%
    -webkit-transform: translateX(-2000px) scale(0.7)
    transform: translateX(-2000px) scale(0.7)
    opacity: 0.7
  50%
    -webkit-transform: translateX(0px) scale(0.7)
    transform: translateX(0px) scale(0.7)
    opacity: 0.7
  100%
    -webkit-transform: scale(1)
    transform: scale(1)
    opacity: 1