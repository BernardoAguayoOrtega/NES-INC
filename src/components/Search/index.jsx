// import react
import React, { useState, useContext } from 'react';
// import context
import { Context } from '../../utils/Context/index.jsx';
// import styles
import './styles.styl';

// create and export Search page
const Search = () => {
  // use state
  const [showSearch, setShowSearch] = useState(false);
  // use state
  const [search, setSearch] = useState('');
  // use context
  const { setAction } = useContext(Context);

  const callAPI = (value) => {
    setAction({ option: 3, action: `'&${value}'`, redirect: true });

    setSearch('')
  };

  return (
    <div>
      <div
        className={`searchformOff ${
          showSearch ? 'slide_search-left searchform' : 'off'
        }`}
      >
        <input
          onKeyPress={(ev) => {
            if (ev.key === 'Enter') {
              callAPI(search)
              ev.preventDefault();
            }
          }}
          value={search}
          onChange={(e) => setSearch(e.target.value)}
          name="search"
          type="search"
          placeholder="Search"
        />
      </div>
      <a className="search" onClick={() => setShowSearch((value) => !value)}>
        <i className="fa fa-search fa-2x" />
      </a>
    </div>
  );
};

export default Search;
