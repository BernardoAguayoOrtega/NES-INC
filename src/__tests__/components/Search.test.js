// import react
import React from 'react';
// import shallow
import { mount } from 'enzyme';
// import toJson
import toJson from 'enzyme-to-json';
// import app component
import Search from '../../components/Search/index.jsx';

describe('<Search />', () => {
  const wrapper = mount(<Search />);

  it('is render the Search?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('is there searchformOff class?', () => {
    expect(wrapper.find('.searchformOff').exists()).toBe(true);
  });

  it('is there off off class name?', () => {
    expect(wrapper.find('.searchformOff').props().className).toBe('searchformOff off');
  });

  it('is there slide left class name?', () => {
    wrapper.find('.search').simulate('click')

    expect(wrapper.find('.searchformOff').props().className).toBe('searchformOff slide_search-left searchform');
  });

  it('matches the snap shot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
