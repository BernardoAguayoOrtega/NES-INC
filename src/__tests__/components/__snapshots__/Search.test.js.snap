// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> matches the snap shot 1`] = `
<Search>
  <div>
    <div
      className="searchformOff slide_search-left searchform"
    >
      <input
        name="search"
        type="search"
      />
    </div>
    <a
      className="search"
      href="#"
      onClick={[Function]}
    >
      <i
        className="fa fa-search fa-2x"
      />
    </a>
  </div>
</Search>
`;
